### ISSUE TITLE
How to get real `sys.stdout`?

### ISSUE THREAD

**Issue by Danon** [pytest-dev/pytest#8775]
I like `@pytest.fixture` idea, it's nice, its extendbile. I like it.

I looked around and I saw `capsys` fixture, it provides an abstraction layer over standard input, output and error, it's nice.

But it's also quite limited, too limited to my needs and I can't use it the way I'd like. In particular, I'd like to capture the output in the test, and not print it to console. Currently, I can't do it.

Secondly, I'd like to get direct access to `sys.stdin` and `sys.stdout`. I noticed, that EVEN IF I DON'T USE `capsys`,`capfd` or other fuxtures, pytest still replaces `sys.stdin` with its own implementation of `io`.

Is there really NO-WAY at all to access the real `sys.stdin`? Or a way not to call `capsys` if it's not used? Or to disable or override the default fixtures?

Basically, is there any way for me to handle `sys.stdin` on my own terms?

#### What's the problem this feature will solve?

Can't access `sys.stdin`, `sys.stdout`, `sys.stderr`. Can't silence the output.

PS: And no, `pytest -s` is not an option for me, I don't want to silence EVERY test, just some.

#### Describe the solution you'd like

Either:
 - So that `capsys` doesn't override `sys.stdin` when it's not used (when no test uses that fixture)
 - Or that I can disable the `capsys` fixture
 - Or that I can retrieve the real `sys.stdin` from `capsys` if it captured it
 - Or anything else that will allow me to extend the default behaviour.

#### Alternative Solutions

Currently, there is now way I can read the `sys.stdin` while using pytest. There's just no way.

**Comment by nicoddemus** [comment]
Hi @Danon,

I think [`sys.__stdout__`](https://docs.python.org/3/library/sys.html#sys.__stdout__) remains untouched by pytest.

**Comment by Danon** [comment]
@nicoddemus Do you perhaps know why `capsys` fixture captures the `sys.stdin` even if I don't use `capsys` in any of my tests?

**Comment by The-Compiler** [comment]
Output capturing is on for all tests - I think it makes sense to do the same for `stdin` as well, so that tests accidentally using `input()` or so result in an error rather than hanging (without any indication of that happening).

**Comment by Danon** [comment]
@The-Compiler Yes, until you want to perform your own handling of `sys.stdin` and `sys.stdout`. Then pytest handlers mess it up :/

**Comment by The-Compiler** [comment]
There's a context manager to disable them, see the very bottom of the [capturing docs](https://docs.pytest.org/en/6.2.x/capture.html).

**Comment by RonnyPfannschmidt** [comment]
@Danon while all this is on - its still not clear what your use case is and what you want to do

there is a number of ways to integrate/configure pytest to leave your io details alone if needed

**Comment by Danon** [comment]
PS: I'm sorry, I'm new to python, don't blame me, please.

> Output capturing is on for all tests - I think it makes sense to do the same for stdin as well, 

But doesn't that break the python rule "explicit is better than implicit"?

**Comment by RonnyPfannschmidt** [comment]
pytest explicitly advertises itself as being able to support print debugging of tests, including isolating the output and giving you useful reports, 

so i think we cover explicit, feel free to open a pr with big red signs if you think it needs more ^^

**Comment by Zac-HD** [comment]
(closing because I think the question has been answered)
